src/lib/pickles/compile.ml

```diff
@@ -360,9 +361,7 @@ struct
            , Ret_value.t )
            Inductive_rule.public_input
       -> auxiliary_typ:(Auxiliary_var.t, Auxiliary_value.t) Impls.Step.Typ.t
-      -> choices:
```

Comment:
Do you mind explain a bit what is happening here?

Reply:
This is the only 'functional' change: instead of passing a function here and then immediately applying the `self` argument that is also given to this function, instead we require the caller to apply it.

This part of the change isn't really the goal itself; what we're actually aiming for is [the change below](https://github.com/MinaProtocol/mina/pull/16849/files#diff-f40ed7a41b5fb04d1384c4f36660ec6f5516ef3acea11633a61599d03f076ad3R1112-R1116). While the inductive rules are still a function, we can't get the length of the returned list; by applying the function, we can directly extract the length and associate it with the `H4_6_with_length.t`'s `'length` parameter, and so we can use that in place of the explicit `branching` argument.